### PR TITLE
C++zip strategy

### PR DESCRIPTION
**Pull request by jicata**
closed Create a C++ Zip Strategy SoftUni-Internal/suls-issues#1908

### CODE REVIEW

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CPlusPlusZipFileExecutionStrategy.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
+﻿namespace OJS.Workers.ExecutionStrategies
+{
+    using System;
+    using System.IO;
+
+    using OJS.Common.Extensions;
+    using OJS.Common.Models;
+
```
Comment (Innos) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CPlusPlusZipFileExecutionStrategy.cs]: Usings in the same primary namespace can be in the same block.

File: Open Judge System/Workers/OJS.Workers.Compilers/CPlusPlusZipCompiler.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                return inputFile;
+            }
+
+            return $"{inputFile}{GlobalConstants.ZipFileExtension}";
+        }
+
+        public override string BuildCompilerArguments(string inputFile, string outputFile, string additionalArguments)
+        {
+            var arguments = new StringBuilder();
+            // Output file argument
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.Compilers/CPlusPlusZipCompiler.cs]: remove

File: Open Judge System/Workers/OJS.Workers.Compilers/CPlusPlusZipCompiler.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            arguments.Append($"-o \"{outputFile}\"");
+            arguments.Append(' ');
+
+            // Additional compiler arguments
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.Compilers/CPlusPlusZipCompiler.cs]: remove

File: Open Judge System/Workers/OJS.Workers.Compilers/CPlusPlusZipCompiler.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            arguments.Append(additionalArguments);
+            arguments.Append(' ');
+            FileHelpers.UnzipFile(inputFile, this.workingDirectory);
+
+            // Input files arguments
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.Compilers/CPlusPlusZipCompiler.cs]: remove

File: Open Judge System/Workers/OJS.Workers.Compilers/CPlusPlusZipCompiler.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            var filesToCompile = Directory.EnumerateFiles(
+                    this.workingDirectory,
+                    "*.*",
+                    SearchOption.AllDirectories)
+                .Where(
+                    f =>
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.Compilers/CPlusPlusZipCompiler.cs]: move on upper row

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CPlusPlusZipFileExecutionStrategy.cs [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    using OJS.Common.Extensions;
+    using OJS.Common.Models;
+    using OJS.Workers.Checkers;
+    using OJS.Workers.Executors;
+
+    public class CPlusPlusZipFileExecutionStrategy : ExecutionStrategy
+    {
+        private const string SubmissionName = "UserSubmission.zip";
+        private readonly Func<CompilerType, string> getCompilerPathFunc;
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CPlusPlusZipFileExecutionStrategy.cs]: insert blank line IMO

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CPlusPlusZipFileExecutionStrategy.cs [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+            DirectoryHelpers.SafeDeleteDirectory(this.WorkingDirectory,true);
+        }
+
+        public string WorkingDirectory { get; set; }
+
+        public override ExecutionResult Execute(ExecutionContext executionContext)
+        {
+            var result = new ExecutionResult();
+
+            string submissionDestination = $@"{this.WorkingDirectory}\{SubmissionName}";
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CPlusPlusZipFileExecutionStrategy.cs]: var

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CPlusPlusZipFileExecutionStrategy.cs [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+                submissionDestination);
+
+            if (!compilationResult.IsCompiledSuccessfully)
+            {
+                return result;
+            }
+
+            result.IsCompiledSuccessfully = true;
+
+           var executor = new RestrictedProcessExecutor();
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CPlusPlusZipFileExecutionStrategy.cs]: tab (you are one space behind)

File: Open Judge System/Workers/OJS.Workers.Compilers/CPlusPlusZipCompiler.cs [hunk 9]
```diff
@@ -39,25 +39,22 @@ public override string RenameInputFile(string inputFile)
...
             arguments.Append($"-o \"{outputFile}\"");
             arguments.Append(' ');
 
-            // Additional compiler arguments
             arguments.Append(additionalArguments);
             arguments.Append(' ');
             FileHelpers.UnzipFile(inputFile, this.workingDirectory);
 
-            // Input files arguments
             var filesToCompile = Directory.EnumerateFiles(
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.Compilers/CPlusPlusZipCompiler.cs]: IMO .EnumeratFiles on new line; one tab; aligned with .Where below; other stuff one more tab to the right

**Pull request merged**